simplesat/transaction.py

```diff
@@ -113,23 +121,40 @@ def _safe_operations(self, pool, decisions, installed_package_ids):
                       package_id not in installed_package_ids):
                     installs.append(package_id)
 
-        # Removals should happen top down
-        for package_id in reversed(removals):
-            package = pool.id_to_package(package_id)
-            operations.append(RemoveOperation(package))
+        install_operations = []
+        remove_operations = []
 
         # Installations should happen bottom up
         for package_id in installs:
-            package = pool.id_to_package(package_id)
-            operations.append(InstallOperation(package))
+            # The solver may suggest removing a currently installed package and
+            # installing another version of that same package from a different
```

Comment:
I haven't gotten it straight in my head yet how we handle identical looking packages from different repositories. We've already said that they aren't equivalent, but we often assume that any attempt to swap one for the other must be a mistake. Do we have the behavior written down anywhere? @noraderam @cournape